Tests/DatadogTests/Datadog/URLSessionAutoInstrumentation/DDURLSessionDelegateAsSuperclassTests.swift

```diff
@@ -0,0 +1,26 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import XCTest
+import Datadog
+
+internal final class SubDDURLSessionDelegate: DDURLSessionDelegate {
+    let someProp: String
+    override init() {
+        someProp = "someProp"
+        super.init()
+    }
+    override func urlSession(_ session: URLSession, task: URLSessionTask, didCompleteWithError error: Error?) {
+        super.urlSession(session, task: task, didCompleteWithError: error)
+    }
+}
+
+class DDURLSessionDelegateAsSuperclassTests: XCTestCase {
```

Comment:
The `DDURLSessionDelegateTests.swift` already defines the region for this kind of tests:
```swift
// MARK: - Usage
```
we should move it there. We test usage this way in other places too, see: `LoggerTests`, `TracerTests`, `RUMMonitorTests`.

Also, for simplicity, the subclass can be defined in test function scope, to not pollute the namespace for only single usage.

Reply:
this must be in a separate file in which `import Datadog` without `@testable`
otherwise it would pass even before the fix as designated initializer was already `internal`

yes, the subclass was supposed to be a `private` or nested class 👌

Reply:
If we remove the unused initializer, as suggested https://github.com/DataDog/dd-sdk-ios/pull/281#discussion_r504823965, this may go into `DDURLSessionDelegateTests`, no?

Reply:
i don't think so becuase the case definition is "can anyone subclass this class without @testable keyword?"

Reply:
I don't see the difference. If the initializer is not defined, how can it be called if `@testable` is used 🤷.

Reply:
i probably didn't understand 🤔 

if one day someone makes some changes and this type becomes unusable in the host app, we hope that this test case will catch that.
if this case had `@testable import Datadog` then we would miss it and this is what just happened: 
i wrote this case in `DDURLSessionDelegateTests` *before* the fix and it passed (but it shouldn't)

Reply:
This should be the part of integration test then - testability is ultimately disabled in their configuration. Anyway, good to go right now, we can improve it later.

Reply:
but this is a test on the unit, basically it tests its public interface only. by definition this is not an integration test to my understanding